src/inference/inference.h

```diff
@@ -67,6 +69,11 @@ class Nice2Assignment {
   virtual void FromJSON(const Json::Value& assignment) = 0;
   virtual void ToJSON(Json::Value* assignment) const = 0;
 
+  virtual void GetCandidates(
+      Nice2Inference& inference,
```

Comment:
Must be a pointer if you change it according to the coding conventions in Nice2Predict.